cwl/src/main/scala/cwl/WorkflowStep.scala

```diff
@@ -206,10 +220,17 @@ case class WorkflowStep(
         3) Use the InputDefinitionFold to build a new call node
        */
       for {
-        stepInputFold <- in.foldLeft(WorkflowStepInputFold.emptyRight)(foldStepInput)
-        inputDefinitionFold <- taskDefinition.inputs.foldMap(foldInputDefinition(stepInputFold.stepInputMapping)).toEither
+        stepInputFold <- stepInputFoldCheck
+        scatterVariables <- ScatterLogic.buildScatterVariables(scatter, stepInputFold, unqualifiedStepId.localName.value)
+        inputDefinitionFold <- taskDefinition.inputs.foldMap(foldInputDefinition(stepInputFold.stepInputMapping, scatterVariables)).toEither
         callAndNodes = callNodeBuilder.build(unqualifiedStepId, taskDefinition, inputDefinitionFold)
-      } yield stepInputFold.generatedNodes ++ callAndNodes.nodes ++ knownNodes
+        allNodes <- if (scatterVariables.nonEmpty) {
+          // NB: Pattern matching the scatterVariables against "head :: tail" to build a NeL doesn't work because the compiler takes :: for the shapeless operator
```

Comment:
you may be able to `import shapeless.{:: => _}` if that helps

Reply:
oh yeah good idea I'll try